util/collections/machine_collection.go

```diff
@@ -176,3 +176,25 @@ func (s Machines) Names() []string {
 	}
 	return names
 }
+
+// SortedByVersion returns the machines sorted by version.
+func (s Machines) SortedByVersion() []*clusterv1.Machine {
```

Comment:
```suggestion
func (s Machines) sortedByVersion() []*clusterv1.Machine {
```
given this is used only in this package